moda-contracts/src/interfaces/ICatalog.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+/**
+ * @title IRegistry
+ * @dev Interface for Registry
+ */
+interface ICatalog {
+    /**
+     * @dev Status of a registered track
+     */
+    enum TrackStatus {
+        PENDING,
+        VALIDATED,
+        INVALIDATED
+    }
+
+    /**
+     * @dev Represents a registered track
+     * TrackStatus - The status of the track (pending, validated, invalidated)
+     * TrackOwner - The address that registered the track
+     * TrackRegistrationHash - This is a IPFS hash of the track metadata
+     * FingerprintHash - This is a IPFS hash of the track fingerprint
+     * ValidationHash - This is a hash of proof of validation data
+     */
+    struct RegisteredTrack {
+        TrackStatus trackStatus;
+        address artist;
+        address trackBeneficiary;
+        string trackRegistrationHash;
+        string fingerprintHash;
+        string validationHash;
+    }
+
+    /**
+     * @dev Represents a registered release
+     * releases - The address of the Releases contract contract
+     * from which The release was created
+     * TokenId - The token id of the release
+     * TrackIds - The ids of the tracks in the release
+     */
+    struct RegisteredRelease {
+        address releases;
+        uint256 tokenId;
+        string[] trackIds;
+    }
+
+    /**
+     * @dev Emitted whenever the track registration hash is updated
+     * as the registration hash is the IPFS hash of the metadata,
+     * this is emitted when calling updateTrackMetadata
+     */
+    event TrackRegistrationHashUpdated(
+        string trackId, string newTrackRegistrationHash, address indexed trackOwner
+    );
+
+    /**
+     * @dev Emitted when a track is registered
+     */
+    event TrackRegistered(string trackRegistrationHash, string trackId, address indexed trackOwner);
+
+    /**
+     * @dev Emitted when the status of a track is set to invalidated
+     */
+    event TrackInvalidated(string trackId);
+
+    /**
+     * @dev Emitted whenever a field in RegisterTrack is updated
+     */
+    event TrackUpdated(
+        TrackStatus indexed trackStatus,
+        address indexed trackArtist,
+        address trackBeneficiary,
+        string trackRegistrationHash,
+        string fingerprintHash,
+        string validationHash
+    );
+
+    /**
+     * @dev Emitted when a track owner has given permission to a
+     * releases contract to be able to release their track
+     */
+    event TrackAddedToReleasesContract(string trackId, address indexed releases);
```

Comment:
Let's drop the word `Contract` on all of these event names

Reply:
done.